frontend/src/features/leases/detail/LeasePages/deposits/DepositsContainer.tsx

```diff
@@ -0,0 +1,257 @@
+  const [deleteReturnModalWarning, setDeleteReturnModalWarning] = useState<boolean>(false);
+
+  const [showReturnEditModal, setShowReturnEditModal] = useState<boolean>(false);
+
+  const [depositToDelete, setDepositToDelete] = useState<FormLeaseDeposit | undefined>(undefined);
+  const [editDepositValue, setEditDepositValue] = useState<FormLeaseDeposit>(
+    FormLeaseDeposit.createEmpty(),
+  );
+
+  const [depositReturnToDelete, setDepositReturnToDelete] = useState<
+    FormLeaseDepositReturn | undefined
+  >(undefined);
+  const [editReturnValue, setEditReturnValue] = useState<FormLeaseDepositReturn | undefined>(
+    undefined,
+  );
+
+  const { updateLeaseDeposit, removeLeaseDeposit } = useLeaseDeposits();
+  const { updateLeaseDepositReturn, removeLeaseDepositReturn } = useLeaseDepositReturns();
+
+  const onAddDeposit = () => {
+    setEditDepositValue(FormLeaseDeposit.createEmpty());
+    setShowEditModal(true);
+  };
+
+  const onEditDeposit = (id: number) => {
+    var deposit = securityDeposits.find(x => x.id === id);
+    if (deposit) {
+      setEditDepositValue(FormLeaseDeposit.createFromModel(deposit));
+      setShowEditModal(true);
+    }
+  };
+
+  const onDeleteDeposit = (id: number) => {
+    var deposit = securityDeposits.find(x => x.id === id);
+    if (deposit) {
+      setDepositToDelete(FormLeaseDeposit.createFromModel(deposit));
+      setDeleteModalWarning(true);
+    }
+  };
+
+  const onReturnDeposit = (id: number) => {
+    var deposit = securityDeposits.find(x => x.id === id);
+    if (deposit) {
+      setEditReturnValue(FormLeaseDepositReturn.createEmpty(deposit));
+      setShowReturnEditModal(true);
+    }
+  };
+
+  const onDeleteDepositConfirmed = async () => {
+    if (lease && lease.id && lease.rowVersion && depositToDelete) {
+      const updatedLease = await removeLeaseDeposit({
+        parentId: lease.id,
+        parentRowVersion: lease.rowVersion,
+        payload: FormLeaseDeposit.toInterfaceModel(depositToDelete),
+      });
+      if (!!updatedLease?.id) {
+        setDepositToDelete(undefined);
+        setDeleteModalWarning(false);
+        setLease(updatedLease);
+      }
+    }
+  };
+
+  const onDeleteDepositReturnConfirmed = async () => {
+    if (lease && lease.id && lease.rowVersion && depositReturnToDelete) {
+      const updatedLease = await removeLeaseDepositReturn({
+        parentId: lease.id,
+        parentRowVersion: lease.rowVersion,
+        payload: FormLeaseDepositReturn.toInterfaceModel(depositReturnToDelete),
+      });
+      if (!!updatedLease?.id) {
+        setDepositReturnToDelete(undefined);
+        setDeleteReturnModalWarning(false);
+        setLease(updatedLease);
+      }
+    }
+  };
+
+  /**
+   * Send the save request (either an update or an add). Use the response to update the parent lease.
+   * @param depositForm
+   */
+  const onSaveDeposit = async (depositForm: FormLeaseDeposit) => {
+    console.log('saving deposit!', depositForm);
```

Comment:
remove.

Reply:
yeah, will remove them once the tests run sucessfuly